packages/flutter/lib/src/material/carousel.dart

```diff
@@ -159,13 +244,16 @@ class CarouselView extends StatefulWidget {
 }
 
 class _CarouselViewState extends State<CarouselView> {
-  late double _itemExtent;
+  double? _itemExtent;
+  List<int>? _weights;
   CarouselController? _internalController;
   CarouselController get _controller => widget.controller ?? _internalController!;
+  late bool allowFullyExpand;
```

Comment:
Why do we need to store a copy of this in the state instead of just accessing it as `widget.allowFullyExpand` everywhere?

Reply:
If this property remains, it should likely be private:

```suggestion
  late bool _allowFullyExpand;
```